File: tests/java/BrainflowGetData.java
import brainflow.*;

public class BrainflowGetData {

    public static void main (String[] args) throws Exception {

        BoardShim.enable_board_logger ();
        BoardShim board_shim = new BoardShim (Integer.parseInt (args[0]), args[1]);
        board_shim.prepare_session ();
        System.out.println ("Session is ready");
        board_shim.start_stream (3600);
        System.out.println ("Started");
        Thread.sleep (10000);
        board_shim.stop_stream ();
        System.out.println ("Stopped");
        System.out.println (board_shim.get_board_data_count ());
        board_shim.release_session ();
        System.out.println ("Released");
    }
}

File: tests/python/brainflow_get_data.py
import argparse
import time
import brainflow
import brainflow.board_shim


def main ():
    parser = argparse.ArgumentParser ()
    parser.add_argument ('--port', type = str, help  = 'port name, for synthetic board port_name doesnt matter, just pass smth', required = True)
    parser.add_argument ('--board', type = int, help  = 'board id, check docs to get a list of supported board ids', required = True)
    parser.add_argument ('--log', action = 'store_true')
    args = parser.parse_args ()

    if (args.log):
        brainflow.board_shim.BoardShim.enable_dev_board_logger ()
    else:
        brainflow.board_shim.BoardShim.disable_board_logger ()

    board = brainflow.board_shim.BoardShim (args.board, args.port)
    board.prepare_session ()
    
    # disable second channel, note emulator doesnt handle such commands, run with real board to validate
    # different board have different data formats
    if args.board == brainflow.board_shim.BoardIds.CYTON.value:
        board.config_board ('x2100000X')
    elif args.board == brainflow.board_shim.BoardIds.GANGLION.value:
        board.config_board ('2')

    board.start_stream ()
    time.sleep (10)
    data = board.get_board_data ()
    board.stop_stream ()
    board.release_session ()

    print (data)

if __name__ == "__main__":
    main ()
